source/tutorials/file-sets.md

```diff
@@ -307,126 +375,168 @@ error: lib.fileset.difference: Second argument (negative set)
   To create a file set from a path that may not exist, use `lib.fileset.maybeMissing`.
 ```
 
-It helpfully explains to us that for files that may not exist,
-we should use `maybeMissing` <!-- https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.fileset.maybeMissing -->,
-so let's try it:
+Follow the instructions in the error message, and use `maybeMissing` to create a file set from a path that may not exist (in which case the file set will be empty):
 
-```{code-block} nix
-:caption: package.nix
-  sourceFiles = fs.difference ./. (fs.maybeMissing ./result);
+<!-- https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.fileset.maybeMissing -->
+
+```{code-block} diff
+:caption: build.nix
+ { stdenv, lib }:
+ let
+   fs = lib.fileset;
+-  sourceFiles = fs.difference ./. ./result;
++  sourceFiles = fs.difference ./. (fs.maybeMissing ./result);
+ in
 ```
 
-This now works, reliably filtering out `./result` if it exists:
+This now works, using the whole directory since `./result` is not present:
 
 ```
 $ nix-build
 trace: /home/user/select (all files in directory)
 this derivation will be built:
-  /nix/store/ygpx17kshzc6bj3c71xlda8szw6qi1sr-filesets.drv
-[...]
-/nix/store/bzvhlr9h2zwqi7rr9i1j193z9hkskhmk-filesets
+  /nix/store/zr19bv51085zz005yk7pw4s9sglmafvn-fileset.drv
+...
+/nix/store/vhyhk6ij39gjapqavz1j1x3zbiy3qc1a-fileset
+```
 
+Another build attempt will produce a different trace, but the same output path:
+
+```
 $ nix-build
-trace: /home/user/select
+trace: /home/user/fileset
+trace: - build.nix (regular)
 trace: - default.nix (regular)
+trace: - hello.txt (regular)
 trace: - nix (all files in directory)
-trace: - package.nix (regular)
-trace: - string.txt (regular)
-/nix/store/bzvhlr9h2zwqi7rr9i1j193z9hkskhmk-filesets
+trace: - world.txt (regular)
+/nix/store/vhyhk6ij39gjapqavz1j1x3zbiy3qc1a-fileset
 ```
 
-## Adding file sets together
+## Union (exclude)
```

Comment:
```suggestion
## Union (for excluding only select files)
```

Though this also makes it sound like that's what this function does, which would be wrong :thinking:

Reply:
```suggestion
## Union (explicitly exclude files)
```

?

Reply:
I like it!